# Individual Planning Project (DSCI 100)

In [1]:
library(tidyverse)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors


## Data Description

This data was collected by a computer science research group lead by Frank Wood at UBC. This information was fromm a MineCraft server they set up to better understand how players interact with the game so that they can train AI to respond and talk to players.

#### 1. Players Dataset

In [2]:
players_data <- read_csv("players.csv")

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


- Number of observations: 196
- Number of variables: 7

Potential issues in data:
- Missing data in Age

In [3]:
players_summary <- players_data |>
    summarize(avg_played_hours = mean(played_hours),
              avg_age = mean(Age, na.rm = TRUE)) |>
    mutate(avg_played_hours = round(avg_played_hours, 2),
           avg_age = round(avg_age, 2))
players_summary

avg_played_hours avg_age
1 5.85             21.14

#### 2. Sessions Dataset

In [5]:
sessions_data <- read_csv("sessions.csv")

Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


- Number of observations: 1535
- Number of variables: 5

Potential issues in data:
- Missing data in end_time and original_end_time
- end_time and start_time will both need to be divided into two seperate columns (date and time) to make the dataset tidy 

In [6]:
sessions_summary <- sessions_data |>
    summarize(avg_start_time = mean(original_start_time),
              avg_end_time = mean(original_end_time, na.rm = TRUE)) |>
    mutate(avg_start_time = round(avg_start_time, 2),
           avg_end_time = round(avg_end_time, 2))
sessions_summary

avg_start_time avg_end_time
1 1.719201e+12   1.719196e+12

## Questions

## Exploratory Data Analysis and Visualization 

## Methods and Plan

## GitHub Repository 